cfy_manager/main.py

```diff
@@ -1211,6 +1216,17 @@ def upgrade(verbose=False, private_ip=None, public_ip=None, config_file=None):
     packages_to_update, _ = _get_packages()
     run(['yum', 'clean', 'all'],
         stdout=sys.stdout, stderr=sys.stderr)
+
+    # handle possible erlang package change:
+    if 'esl-erlang' in packages_to_update and is_package_installed('erlang'):
+        packages_to_update.remove('esl-erlang')
+        erlang_pkg = run(['rpm', '-q', 'erlang']).aggr_stdout.strip()
+        run(['rpm', '-e', '--nodeps', erlang_pkg],
+            stdout=sys.stdout, stderr=sys.stderr)
+        run(['yum', 'install', '-y', '--disablerepo=*',
+             '--enablerepo=cloudify', 'esl-erlang'],
+            stdout=sys.stdout, stderr=sys.stderr)
```

Comment:
this needs to be just... better quality
- is this actually needed? the comment says "possible package change". What change? Possible? Are we not sure if it changed?
- does erlang actually need to be removed?
- feel free to separate this into its own function
- `rpm -e` rather than `yum_remove`?
- we have `yum_install` for that yum install call

Reply:
cool, would be good to put that knowledge in the comments right there in the code (hence suggesting a separate function, where we can comment all we want, without putting too much cruft in the `upgrade` func)